tests/tests.rs

```diff
@@ -395,6 +395,26 @@ sherlock!(csglob, "Sherlock", ".", |wd: WorkDir, mut cmd: Command| {
     assert_eq!(lines, "file2.html:Sherlock\n");
 });
 
+sherlock!(byte_offset, "Sherlock", ".", |wd: WorkDir, mut cmd: Command| {
+    cmd.arg("-b");
+    let lines: String = wd.stdout(&mut cmd);
+    let expected = "\
+sherlock:0:For the Doctor Watsons of this world, as opposed to the Sherlock
+sherlock:129:be, to a very large extent, the result of luck. Sherlock Holmes
+";
+    assert_eq!(lines, expected);
+});
+
+sherlock!(byte_offset_only_matching, "Sherlock", ".", |wd: WorkDir, mut cmd: Command| {
+    cmd.arg("-b").arg("-o");
+    let lines: String = wd.stdout(&mut cmd);
+    let expected = "\
+sherlock:56:Sherlock
+sherlock:177:Sherlock
+";
+    assert_eq!(lines, expected);
+});
+
```

Comment:
For completeness, could you pass the `--no-mmap` flags to each of the above tests, and then copy the tests and change `--no-mmap` to `--mmap`? I am thinking that your implementation might actually be correct for `--mmap` but not for `--no-mmap`. I'm not sure a global byte offset is even tracked in the `--no-mmap` searcher, so this task might require a bit more work. :)

Reply:
I'm really sorry about overlooking this. When I went into `search_stream`, I noticed that the change was pretty trivial and it made so much sense to me that I missed `search_buffer`. It's on me. I'll have a look at the `mmap` searcher to see what can be done. :+1:

Reply:
Ah, I think I lucked out here. The iterator returned by `self.grep.iter(self.buf)` in `search_buffer.rs` provides us with the starting position of the matching line itself. From then on, we call `printer.matched` with the `start` and `end`, which thankfully works the same way it does for `search_stream`. Both the `--no-mmap` and `--mmap` tests that you asked to write also pass. :+1:

Reply:
Hmm, awesome! You might actually want to test this also in non-mmap searcher using a smaller capacity, such that the matches reported span multiple buffers. If that test passes, then I'll believe the implementation is correct. :)

Reply:
I believe there's a `smallcap` helper function in the tests somewhere, since there are other tests that proceed in a similar vein. But I can't remember. It's been a while since I looked at that code.

Reply:
Hmm... No, you're right. I tried it with the `smallcaps` test available in `search_stream.rs` and it completely blew apart. (I got 0 as the offset for the next match instead of the offset from the beginning of the file.) The good news is that I can see why this is happening very clearly.

I now understand why you originally thought it will require additional work for the `--no-mmap` case. I'll have a look at it. Sorry. :+1: